controllers/selfnoderemediation_controller.go

```diff
@@ -164,8 +181,51 @@ func (r *SelfNodeRemediationReconciler) Reconcile(ctx context.Context, req ctrl.
 	return result, r.updateSnrStatusLastError(snr, err)
 }
 
+func (r *SelfNodeRemediationReconciler) updateSnrProcessingCondition(reason processingChangeReason, snr *v1alpha1.SelfNodeRemediation, isPatchDone bool) error {
```

Comment:
The signature of this function makes it a bit cryptic when used. I need to look at its definition to know why it's `updateSnrProcessingCondition(..., true)` or `updateSnrProcessingCondition(..., false)`.
It seems it's doing two things: update the processing condition AND (sometimes) patching SNR status, what do you think about having `updateSnrProcessingCondition` and `updateSnrProcessingConditionAndPatch` which calls the first internally?

Reply:
Also, the `DeepCopy` is always done regardless it needs to patch or not.

Reply:
TBH I don't have a strong preference but I see your  point.
I think that if going in that direction a nicer solution would be just to separate the patch from the status change.
let me know if you are ok with this change.